classes/components/forms/context/DoiSetupSettingsForm.inc.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * @file classes/components/form/context/DoiSettingsForm.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DoiSettingsForm
+ * @ingroup classes_controllers_form
+ *
+ * @brief A preset form for enabling and configuring DOI settings for a given context
+ */
+
+namespace APP\components\forms\context;
+
+use APP\facades\Repo;
+use PKP\components\forms\context\PKPDoiSetupSettingsForm;
+use PKP\components\forms\FieldOptions;
+use PKP\components\forms\FieldSelect;
+use PKP\components\forms\FieldText;
+use PKP\context\Context;
+
+class DoiSetupSettingsForm extends PKPDoiSetupSettingsForm
+{
+    public function __construct(string $action, array $locales, Context $context)
+    {
+        parent::__construct($action, $locales, $context);
+
+        // Added in PKPDoiSetupSettingsForm, but not currently applicable to OMP
+        $this->removeField(Context::SETTING_DOI_AUTOMATIC_DEPOSIT);
+
+        $this->addField(new FieldOptions(Context::SETTING_ENABLED_DOI_TYPES, [
+            'label' => __('doi.manager.settings.doiObjects'),
+            'description' => __('doi.manager.settings.doiObjectsRequired'),
+            'options' => [
+                [
+                    'value' => Repo::doi()::TYPE_PUBLICATION,
+                    'label' => __('doi.manager.settings.enableFor', ['objects' => __('common.publication')]),
+                ],
+                [
+                    'value' => Repo::doi()::TYPE_CHAPTER,
+                    'label' => __('doi.manager.settings.enableFor', ['objects' => __('submission.chapter')]),
+                ],
+                [
+                    'value' => Repo::doi()::TYPE_REPRESENTATION,
+                    'label' => __('doi.manager.settings.enableFor', ['objects' => __('monograph.publicationFormats')]),
+                ],
+                [
+                    'value' => Repo::doi()::TYPE_SUBMISSION_FILE,
+                    'label' => __('doi.manager.settings.enableFor', ['objects' => __('plugins.pubIds.doi.manager.settings.enableSubmissionFileDoi')]),
+                ],
+            ],
+            'value' => $context->getData(Context::SETTING_ENABLED_DOI_TYPES) ? $context->getData(Context::SETTING_ENABLED_DOI_TYPES) : [],
+            'showWhen' => Context::SETTING_ENABLE_DOIS
+        ]))
+            ->addField(new FieldText(Context::SETTING_DOI_PREFIX, [
+                'label' => __('doi.manager.settings.doiPrefix'),
+                'description' => __('doi.manager.settings.doiPrefix.description'),
+                'value' => $context->getData(Context::SETTING_DOI_PREFIX),
+                'showWhen' => Context::SETTING_ENABLE_DOIS,
+                'size' => 'small',
+                'isRequired' => true
+
+            ]))
```

Comment:
I think this field can be moved to `PKPDoiSetupSettingsForm`.

Reply:
Same with `Context::SETTING_DOI_CREATION_TIME` and `Context::SETTING_CUSTOM_DOI_SUFFIX_TYPE`.